File: justorder_framework/src/main/java/it/polito/justorder_framework/model/Restaurant.java
package it.polito.justorder_framework.model;

import android.net.Uri;

import java.util.HashMap;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Restaurant {
    private String key;
    private String name;
    private Uri imageUri;
    private String telephone;
    private String address;
    private String fiscalCode;
    private String vatCode;
    private String iban;
    private String openingHour;
    private String closingHour;
    private List<String> openingDays;
    private List<String> type;

    private String owner;
    private HashMap<String, Boolean> managers;
}

File: justorder_framework/src/main/java/it/polito/justorder_framework/model/Restaurant.kt
package it.polito.justorder_framework.model

import java.io.Serializable

class Restaurant : Model(),Serializable {
    var name: String = ""
    var email: String = ""
    var imageUri: String? = null
    var telephone: String = ""
    var address: String = ""
    var fiscalCode: String = ""
    var vatCode: String = ""
    var iban: String = ""
    var openingHour: String = ""
    var closingHour: String = ""
    var openDays: Map<String, Boolean> = mutableMapOf()
    var type: String = ""

    var products: Map<String, Product> = mutableMapOf()
    var orders: Map<String, Boolean> = mutableMapOf<String, Boolean>()

    var owner: String = ""

    var messagingToken : String? = null
}
